vividus-plugin-web-app-playwright/src/main/java/org/vividus/ui/web/playwright/reporter/RequestsCollectingStoryReporter.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2019-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.ui.web.playwright.reporter;
+
+import org.jbehave.core.model.Meta;
+import org.jbehave.core.model.Scenario;
+import org.jbehave.core.model.Story;
+import org.jbehave.core.steps.Timing;
+import org.vividus.ChainedStoryReporter;
+import org.vividus.context.RunContext;
+
+public class RequestsCollectingStoryReporter extends ChainedStoryReporter
+{
+    private static final String COLLECT_REQUESTS_META = "collectRequests";
+
+    private final RunContext runContext;
+    private final boolean requestsCollectingEnabled;
+
+    private boolean collectingEnabled;
+
+    public RequestsCollectingStoryReporter(RunContext runContext, boolean requestsCollectingEnabled)
+    {
+        this.runContext = runContext;
+        this.requestsCollectingEnabled = requestsCollectingEnabled;
+    }
+
+    @Override
+    public void beforeStory(Story story, boolean givenStory)
```

Comment:
if it's possible I would recommend to avoid using `StoryReporter` here, but rather use `@BeforeStory/Scenario` and `@AfterStory/Scenario` hooks

Reply:
> if it's possible I would recommend to avoid using `StoryReporter` here, but rather use `@BeforeStory/Scenario` and `@AfterStory/Scenario` hooks

You're about annotations? Its only work for Jbehave steps

Reply:
On the other hand, why we shouldn't register the class with monitor as steps.